R/FileBlock.R

```diff
@@ -0,0 +1,50 @@
+#' @title `FileBlock`
+#' @keywords internal
+#'
+FileBlock <- R6::R6Class( # nolint: object_name_linter.
+  classname = "FileBlock",
+  inherit = ContentBlock,
+  public = list(
+    #' @description finalize of this `FileBlock`.
+    #'
+    #' @details Removes the temporary file created in the constructor.
+    #'
+    finalize = function() {
+      try(unlink(super$get_content()))
+    },
+    #' @description Create the `FileBlock` from a list.
+    #' The list should contain one named field, `"basename"`.
+    #' @param x `named list` with one field `"basename"`, a name of the file.
+    #' @param output_dir `character` with a path to the dir where a file will be copied.
+    #' @return invisibly self
+    #' @examples
+    #' block <- teal.reporter:::FileBlock$new()
+    #' file_path <- tempfile(fileext = ".png")
+    #' saveRDS(iris, file_path)
+    #' block$from_list(list(path = file_path))
+    from_list = function(x, output_dir) {
+      checkmate::assert_list(x)
+      checkmate::assert_names(names(x), must.include = "basename")
+      path <- file.path(output_dir, x$basename)
+      file_type <- paste0(".", tools::file_ext(path))
+      checkmate::assert_file_exists(path, extension = file_type)
+      new_file_path <- tempfile(fileext = file_type)
+      file.copy(path, new_file_path)
+      super$set_content(new_file_path)
```

Comment:
Very important part.
When the FileBlock is finalized then its content path is removed so we have to be careful here.
We have to use an external files for the PictureBlock and TableBlock content paths. If we add here the archive paths then we will break the process, as then the archive files will be removed when the blocks will be removed.